GrpcClient/EmployeeGrpcRepository.cs

```diff
@@ -0,0 +1,286 @@
+﻿using Grpc.Core;
+using Grpc.Net.Client;
+using RepositoryContracts;
+
+namespace GrpcClient;
+
+public class EmployeeGrpcRepository : IEmployeeRepository
+{
+    private string _grpcAddress { get; set; }
+
+    public EmployeeGrpcRepository()
+    {
+        _grpcAddress = "http://192.168.140.143:50051";
+    }
+
+    public async Task<Entities.Employee> AddAsync(Entities.Employee employee)
+    {
+        using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+        //create client for employee entity, "EmployeeClient" method is auto-genereted               
+        var client = new Employee.EmployeeClient(channel);
+        //reply - "AddSingleEmployee(newEmployeeDto)" comes from .proto file + need to convert the og-DTO to .proto-DTO
+        try
+        {
+            var reply = await client.AddSingleEmployeeAsync(new NewEmployeeDTO
+            {
+                FirstName = employee.FirstName,
+                LastName = employee.LastName,
+                WorkingNumber = UInt32.CreateChecked(employee.WorkingNumber),
+                Email = employee.Email,
+                Password = employee.Password
+            });
+            //converting
+            List<ShiftDTO> grpcShiftDtos = reply.AssignedShifts.Dtos.ToList();
+            List<Entities.Shift> shifts = GrpcShiftDtosToEntityShiftsList(grpcShiftDtos);
+            Entities.Employee employeeRecieved = grpcEmployeeDtoToEntityEmployee(reply, shifts);
+        
+            return employeeRecieved;
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.AlreadyExists)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+
+
+
+
+        return null;
+    }
+
+    public async Task<Entities.Employee> UpdateAsync(Entities.Employee employee)
+    {
+        Entities.Employee employee2 = new Entities.Employee();
```

Comment:
Why is it Entities.Employee employee in parenthesis and Entities.Employee employee2 inside the method? And how this method works in general?

Reply:
it returns task of employee and also employee is a parameter in this method, "entities.employee" means that we are using employe from csharp and the grpc(java) one